json_annotation/lib/src/json_serializable.dart

```diff
@@ -50,3 +52,24 @@ class JsonKey {
       : this.nullable = nullable ?? true,
         this.includeIfNull = includeIfNull;
 }
+
+// TODO(kevmoo): Add documentation
+abstract class JsonMapWrapper extends MapBase<String, dynamic> {
+  /// This operation is not supported by an unmodifiable map.
```

Comment:
Hmmm, is there a mixin for this? Otherwise this has to keep up to date with any new apis that the sdk adds....

Also, I think this should throw on `putIfAbsent` and `addAll` as well.

Reply:
Dude! WIP!

MapBase handles everything but what's overloaded here – it's nice.

Reply:
Ah ok looking at the docs for `MapBase` this should be fine.